src/hotspot/share/opto/block.cpp

```diff
@@ -620,10 +620,19 @@ static bool no_flip_branch(Block *b) {
 // fake exit path to infinite loops.  At this late stage they need to turn
 // into Goto's so that when you enter the infinite loop you indeed hang.
 void PhaseCFG::convert_NeverBranch_to_Goto(Block *b) {
-  // Find true target
+  // NeverBranch sits at end_idx, the two projections right after it
   int end_idx = b->end_idx();
-  int idx = b->get_node(end_idx+1)->as_Proj()->_con;
-  Block *succ = b->_succs[idx];
+  int taken_idx = b->get_node(end_idx+1)->as_Proj()->_con;
+  ProjNode* alwaysTaken = b->get_node(end_idx + 1 + taken_idx)->as_Proj();
```

Comment:
I find this code rather confusing. Since it's guaranteed that `alwaysTaken->_con == 0`, can't we simply do something like this?
```
ProjNode* alwaysTaken = b->get_node(end_idx)->as_MultiBranch()->proj_out(0);
Block* succ ==  get_block_for_node(alwaysTaken->unique_ctrl_out_or_null());
```

Reply:
:+1: